### PR TITLE
[FIXED] Issues/530 death sound too loud

### PR DESCRIPTION
**Pull request by DanielFredriksson**
Also fixes water impact sound being too loud when a player died.

### CODE REVIEW

**Comment by viktor4006094** [comment]
bug still encountered in `Debug` configuration when dying of insanity

File: Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp [hunk 1]
```diff
@@ -342,6 +367,104 @@ void AudioSystem::updateProjectileLowPass(Audio::SoundInfo_General* general) {
 	m_audioEngine->updateProjectileLowPass(general->frequency, general->soundID);
 }
 
+void AudioSystem::dealWithDeathSound(AudioComponent* audioC, float dt) {
+	Audio::SoundInfo_General* soundGeneral = &audioC->m_sounds[Audio::SoundType::DEATH];
+	if (soundGeneral->isPlaying) {
+		int soundPoolSize = audioData.m_soundsUnique[Audio::SoundType::DEATH].size();
+
+		if (soundPoolSize > 0) {
+			if (soundGeneral->isPlaying) {
```
Comment (Smaugmuel) [inline Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp]: Double check in same nested if-statement

File: Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp [hunk 2]
```diff
@@ -342,6 +367,104 @@ void AudioSystem::updateProjectileLowPass(Audio::SoundInfo_General* general) {
...
+	}
+}
+
+void AudioSystem::dealwithInsanitySound(AudioComponent* audioC, float dt) {
+	Audio::SoundInfo_General* soundGeneral = &audioC->m_sounds[Audio::SoundType::INSANITY_SCREAM];
+	if (soundGeneral->isPlaying) {
+		int soundPoolSize = audioData.m_soundsUnique[Audio::SoundType::INSANITY_SCREAM].size();
+
+		if (soundPoolSize > 0) {
+			if (soundGeneral->isPlaying) {
```
Comment (Smaugmuel) [inline Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp]: Double check in same nested if-statement

File: Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp (hunk 1 above)
Comment (DanielFredriksson) [reply Sail/src/Sail/entities/systems/Audio/AudioSystem.cpp]: Fixed

**Pull request merged**